app/services/katello/pulp3/repository/file.rb

```diff
@@ -24,6 +24,10 @@ def remotes_api
           PulpFileClient::RemotesFileApi.new(api_client)
         end
 
+        def self.remotes_api(smart_proxy)
+          PulpFileClient::RemotesFileApi.new(api_client(smart_proxy))
+        end
```

Comment:
this looks great!   Can you change the instance 'remotes_api' method to just call this class method, and move it to r  epository.rb  class?   And then also make this change for the other types and also do the same thing for distributions?